soh/soh/Enhancements/Autosave.cpp

```diff
@@ -0,0 +1,77 @@
+#include <libultraship/bridge.h>
+#include "soh/Enhancements/game-interactor/GameInteractor.h"
+#include "soh/Notification/Notification.h"
+#include "soh/ShipInit.hpp"
+#include "soh/SaveManager.h"
+
+extern "C" {
+extern PlayState* gPlayState;
+#include "functions.h"
+#include "variables.h"
+}
+
+static uint64_t lastSaveTimestamp = GetUnixTimestamp();
+
+#define CVAR_AUTOSAVE_NAME CVAR_ENHANCEMENT("Autosave")
+#define CVAR_AUTOSAVE_DEFAULT AUTOSAVE_OFF
+#define CVAR_AUTOSAVE_VALUE CVarGetInteger(CVAR_AUTOSAVE_NAME, CVAR_AUTOSAVE_DEFAULT)
+#define THREE_MINUTES_IN_UNIX 3 * 60000
+
+typedef enum {
+    AUTOSAVE_OFF,
+    AUTOSAVE_ON,
+} AutosaveOptions;
+
+bool Autosave_CanSave() {
+
+    // Don't save when in title screen
+    // Don't save in Ganon's fight and chamber of sages because of master sword and remember save location issues.
+    if (!GameInteractor::IsSaveLoaded(true) || gPlayState->gameplayFrames < 60 ||
```

Comment:
what's the logic/reasoning behind `gPlayState->gameplayFrames < 60` here? i know `IsSaveLoaded`is handling the title screen check, and the `sceneNum` checks are handling the "Don't save in Ganon's fight and chamber of sages" part, but it's not clear what the check for having 3 seconds of game time is doing

Reply:
When entering a save, the magic bar animates from empty to the previous magic fill level. Waiting the 60 frames ensures it doesn't auto save while the magic bar is refilling.

Reply:
ah yeah, that makes sense

i remembered https://github.com/HarbourMasters/Shipwright/pull/3389 existing, but i had forgotten about https://github.com/HarbourMasters/Shipwright/pull/1792

Reply:
Ooooh that makes sense. I'll add that to the comment when I got time then.